controllers/opentelemetrycollector_controller.go

```diff
@@ -50,6 +51,8 @@ import (
 	"github.com/open-telemetry/opentelemetry-operator/pkg/featuregate"
 )
 
+const collectorTargetAllocatorLabelName = "opentelemetry.io/target-allocator"
```

Comment:
should we put this [here](https://github.com/open-telemetry/opentelemetry-operator/blob/main/pkg/constants/env.go)?